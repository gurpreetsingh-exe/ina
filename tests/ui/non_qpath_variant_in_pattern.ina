// COMPILE_FAIL

type T = A | B

fn main() {
    match T::A {
        A => {},
        B => {},
    }
}
