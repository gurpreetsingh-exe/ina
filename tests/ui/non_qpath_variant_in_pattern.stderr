error: pattern binding `A` is named the same as one of the variants of the type `T`
   ╭─[tests/ui/non_qpath_variant_in_pattern.ina:7:9]
   │
 7 │         A => {},
   │         ┬
   │         ╰─ use qualified path `T::A`
   ∙
───╯
error: pattern binding `B` is named the same as one of the variants of the type `T`
   ╭─[tests/ui/non_qpath_variant_in_pattern.ina:8:9]
   │
 8 │         B => {},
   │         ┬
   │         ╰─ use qualified path `T::B`
   ∙
───╯
